modules/soc_mypeugeot/main.sh

```diff
@@ -18,19 +31,135 @@ case $CHARGEPOINT in
 		# defaults to first charge point for backward compatibility
 		# set CHARGEPOINT in case it is empty (needed for logging)
 		CHARGEPOINT=1
+		manualSocFile="$RAMDISKDIR/manual_soc_lp1"
+		manualMeterFile="$RAMDISKDIR/manual_soc_meter_lp1"
+		socFile="$RAMDISKDIR/soc"
 		soctimerfile="$RAMDISKDIR/soctimer"
+		socIntervall=1 # update every minute if script is called every 10 seconds
+		meterFile="$RAMDISKDIR/llkwh"
+		ladungaktivFile="$RAMDISKDIR/ladungaktivlp1"
+		akkug=$akkuglp1
+		efficiency=$wirkungsgradlp1
 		username=$mypeugeot_userlp1
 		password=$mypeugeot_passlp1
 		clientId=$mypeugeot_clientidlp1
 		clientSecret=$mypeugeot_clientsecretlp1
 		;;
 esac
 
-timer=$(<$soctimerfile)
-if (( timer < 60 )); then
-	timer=$((timer+1))
-	echo $timer > $soctimerfile
-else
-	echo 0 > $soctimerfile
-	sudo python $MODULEDIR/peugeotsoc.py $CHARGEPOINT $username $password $clientId $clientSecret
-fi
+socDebugLog(){
+	if (( socDebug > 0 )); then
+		timestamp=`date --rfc-3339=seconds`
+		echo "$timestamp: Lp$CHARGEPOINT: $@" >> $LOGFILE
+	fi
+}
+
+incrementTimer(){
+	soctimer=$((soctimer+1))
+	echo $soctimer > $soctimerfile
+}
+
+soctimer=$(<$soctimerfile)
+
+if (($mypeugeot_soccalclp1 == 0)); then #manual calculation not enabled, using existing logic
```

Comment:
Bitte je nach Ladepunkt prüfen! Mit dieser Logik wird auch an LP2 die Einstellung von LP1 verwendet.
Dazu in der Case-Abfrage zu Beginn eine neue Variable `$soccalc` definieren und je nach Ladepunkt mit `$mypeugeot_soccalclp1` oder `$mypeugeot_soccalclp2` initialisieren.